iceoryx_examples/singleprocess/single_process.cpp

```diff
@@ -44,39 +45,57 @@ void consoleOutput(const std::string& output)
 
 void sender()
 {
-    iox::popo::Publisher publisher({"Single", "Process", "Demo"});
+    iox::popo::UntypedPublisher publisher({"Single", "Process", "Demo"});
     publisher.offer();
 
     uint64_t counter{0};
     while (keepRunning.load())
     {
-        auto sample = static_cast<TransmissionData_t*>(publisher.allocateChunk(sizeof(TransmissionData_t)));
-        sample->counter = counter++;
-        consoleOutput(std::string("Sending: " + std::to_string(sample->counter)));
-        publisher.sendChunk(sample);
+        publisher.loan(sizeof(TransmissionData_t)).and_then([&](iox::popo::Sample<void>& sample) {
+            auto rawSample = static_cast<TransmissionData_t*>(sample.get());
+            rawSample->counter = counter++;
+            consoleOutput(std::string("Sending: " + std::to_string(rawSample->counter)));
+            sample.publish();
+        });
 
         std::this_thread::sleep_for(std::chrono::milliseconds(100));
     }
 }
 
 void receiver()
 {
-    iox::popo::Subscriber subscriber({"Single", "Process", "Demo"});
+    iox::popo::UntypedSubscriber subscriber({"Single", "Process", "Demo"});
 
     uint64_t cacheQueueSize = 10;
     subscriber.subscribe(cacheQueueSize);
 
     while (keepRunning.load())
     {
-        if (iox::popo::SubscriptionState::SUBSCRIBED == subscriber.getSubscriptionState())
+        if (iox::SubscribeState::SUBSCRIBED == subscriber.getSubscriptionState())
         {
-            const void* rawSample = nullptr;
-            while (subscriber.getChunk(&rawSample))
+            bool hasMoreSamples{true};
+            const TransmissionData_t* receivedSample{nullptr};
+
+            do
             {
-                auto sample = static_cast<const TransmissionData_t*>(rawSample);
-                consoleOutput(std::string("Receiving : " + std::to_string(sample->counter)));
-                subscriber.releaseChunk(rawSample);
-            }
+                subscriber.take()
+                    .and_then([&](iox::popo::Sample<const void>& sample) {
+                        receivedSample = static_cast<const TransmissionData_t*>(sample.get());
+                        consoleOutput(std::string("Receiving : " + std::to_string(receivedSample->counter)));
+                    })
+                    .if_empty([&] { hasMoreSamples = false; });
```

Comment:
what if `take()` continuously returns an error? There would be an infinite loop

Reply:
something like
```
subscriber.getNewSamples([] (const auto& sample) {
    // do something with the sample
});
```
where the lambda is called for each new sample

Reply:
Do I have an ara::com Deja Vu?

Reply:
No, it's just a glitch in the Matrix :D

Reply:
@elBoberido to be a little bit consistent with the STL naming I would call such a method `forEachNewSample` and use it as you described it.

Reply:
Added error handling.